File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let left = 0
    let right = height.length - 1
    let max = 0

    while (left < right) {
        let containerHeight = Math.min(height[left], height[right])
        let width = right - left
        let water = containerHeight * width
        max = Math.max(max, water)
        if (height[left] < height[right]) {
            left++
        } else {
            right--
        }
    }

    return max
};

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        max_water = 0

        while left < right:
            min_height = min(height[left], height[right])
            water = min_height * (right - left)
            max_water = max(max_water, water)

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
            
        return max_water